storage/app/src/main/java/com/google/firebase/referencecode/storage/KotlinStorageActivity.kt

```diff
@@ -0,0 +1,405 @@
+package com.google.firebase.referencecode.storage
+
+import android.graphics.Bitmap
+import android.graphics.drawable.BitmapDrawable
+import android.net.Uri
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.widget.ImageView
+import com.google.android.gms.tasks.Continuation
+import com.google.android.gms.tasks.Task
+import com.google.firebase.referencecode.storage.interfaces.StorageActivityInterface
+import com.google.firebase.storage.FirebaseStorage
+import com.google.firebase.storage.StorageMetadata
+import com.google.firebase.storage.StorageReference
+import com.google.firebase.storage.UploadTask
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FileInputStream
+
+class KotlinStorageActivity : AppCompatActivity(), StorageActivityInterface {
+    // [START storage_field_declaration]
+    // [END storage_field_declaration]
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_storage)
+
+        // [START storage_field_initialization]
+        val storage = FirebaseStorage.getInstance()
+        // [END storage_field_initialization]
+
+        includesForCreateReference()
+    }
+
+    override fun includesForCreateReference() {
+        val storage = FirebaseStorage.getInstance()
+
+        // ## Create a Reference
+
+        // [START create_storage_reference]
+        // Create a storage reference from our app
+        var storageRef = storage.reference
+        // [END create_storage_reference]
+
+        // [START create_child_reference]
+        // Create a child reference
+        // imagesRef now points to "images"
+        var imagesRef: StorageReference? = storageRef.child("images")
+
+        // Child references can also take paths
+        // spaceRef now points to "images/space.jpg
+        // imagesRef still points to "images"
+        var spaceRef = storageRef.child("images/space.jpg")
+        // [END create_child_reference]
+
+        // ## Navigate with References
+
+        // [START navigate_references]
+        // getParent allows us to move our reference to a parent node
+        // imagesRef now points to 'images'
+        imagesRef = spaceRef.parent
+
+        // getRoot allows us to move all the way back to the top of our bucket
+        // rootRef now points to the root
+        val rootRef = spaceRef.root
+        // [END navigate_references]
+
+        // [START chain_navigation]
+        // References can be chained together multiple times
+        // earthRef points to 'images/earth.jpg'
+        val earthRef = spaceRef.parent?.child("earth.jpg")
```

Comment:
This doesn't match the java snippet, should be `imagesRef?.child("earth.jpg")` for consistency.  Or did you have a reason for the change?

Reply:
The current java snippet is: `StorageReference earthRef = spaceRef.getParent().child("earth.jpg");`.
This is supposed to show how to chain references together. Simply calling `imagesRef?.child("earth.jpg")` doesn't really illustrate the chaining technique.

Reply:
Oops my mistake!